core/src/main/java/hudson/cli/BuildCommand.java

```diff
@@ -134,9 +134,6 @@ protected int run() throws Exception {
 
                 // not passed in use default
                 ParameterValue defaultValue = pd.getDefaultParameterValue();
-                if (defaultValue == null) {
```

Comment:
It doesn't sound good to completely remove the check as it could cause other issues. At least it should check whether the parameter is optional before doing this.

Reply:
I agree. I don't know how to fix this though. I don't know if we can check if the parameter is optional or if we can whitelist the *hudson.model.FileParameterDefinition* parameter.

Reply:
I would think that if a parameter is optional, is has a default value. If that is not the case, then the build process has more logic and that is not something the core could check. 
@placaze If you need to not fail in case there is no default value, maybe you could use a default value in your configuration and test it in your build process.

However, I don't think this is the correct way to achieve what I understand from your problem, as it could do more damage.